pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -61,21 +74,31 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-// blank assignment to verify that ReconcileUserSignup implements reconcile.Reconciler
 var _ reconcile.Reconciler = &ReconcileUserSignup{}
 
+func NewSignupError(msg string) SignupError {
+	return SignupError{message: msg}
+}
+
+type SignupError struct {
+	message string
+}
+
+func (err SignupError) Error() string {
+	return err.message
+}
+
 // ReconcileUserSignup reconciles a UserSignup object
 type ReconcileUserSignup struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
 	client client.Client
 	scheme *runtime.Scheme
+	clientset kubernetes.Interface
```

Comment:
Do we really need to introduce this `clientset`, as we already have client available?